homeassistant/components/pglab/__init__.py

```diff
@@ -0,0 +1,87 @@
+"""PG LAB Electronics integration."""
+from __future__ import annotations
+
+from pypglab.mqtt import Client, Sub_State, Subcribe_CallBack
+
+from homeassistant.components import mqtt
+from homeassistant.components.mqtt import ReceiveMessage
+from homeassistant.components.mqtt.subscription import (
+    async_prepare_subscribe_topics,
+    async_subscribe_topics,
+    async_unsubscribe_topics,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import (
+    DEVICE_ALREADY_DISCOVERED,
+    DISCONNECT_COMPONENT,
+    DISCOVERY_INSTANCE,
+    PLATFORMS,
+)
+from .discovery import CreateDiscovery
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up PG LAB Electronics integration from a config entry."""
+
+    # define the call back for pglab  module to publish a mqtt message
+    async def mqtt_publish(
+        topic: str, payload: str, qos: int | None = 0, retain: bool | None = False
+    ) -> None:
+        await mqtt.async_publish(hass, topic, payload, qos, retain)
+
+    # define the call back for pglab module to subscribe to a mqtt message
+    async def mqtt_subscribe(
```

Comment:
Same comments as for `mqtt_publish`

Reply:
Please see previous comment.

Reply:
I added the doc string as you suggested in your today review... please reopen if I miss something.